apps/inheritance_app/inheritance_auth_wallet.c

```diff
@@ -118,32 +118,49 @@ static bool verify_auth_wallet_inputs() {
 }
 
 static bool auth_wallet_get_entropy() {
-  secure_data_t msgs[1] = {0};
-  msgs[0].plain_data_size = WALLET_ID_SIZE;
-  memcpy(msgs[0].plain_data, auth->wallet_id, WALLET_ID_SIZE);
-
-  card_error_type_e status = card_fetch_encrypt_data(auth->wallet_id, msgs, 1);
-
-  delay_scr_init(ui_text_inheritance_wallet_authenticating, DELAY_SHORT);
-
-  if (status != CARD_OPERATION_SUCCESS ||
-      msgs[0].encrypted_data_size > ENTROPY_SIZE_LIMIT) {
-    return false;
+  if (!auth->is_seed_based) {
+    secure_data_t msgs[1] = {0};
+    msgs[0].plain_data_size = WALLET_ID_SIZE;
+    memcpy(msgs[0].plain_data, auth->wallet_id, WALLET_ID_SIZE);
+
+    card_error_type_e status =
+        card_fetch_encrypt_data(auth->wallet_id, msgs, 1);
+    if (status != CARD_OPERATION_SUCCESS ||
+        msgs[0].encrypted_data_size > ENTROPY_SIZE_LIMIT) {
+      inheritance_send_error(ERROR_COMMON_ERROR_CORRUPT_DATA_TAG,
+                             ERROR_DATA_FLOW_INVALID_DATA);
+      delay_scr_init(ui_text_inheritance_wallet_auth_fail, DELAY_TIME);
+      return false;
+    }
+    memcpy((void *)auth->entropy,
+           msgs[0].encrypted_data,
+           msgs[0].encrypted_data_size);
+    auth->entropy_size = msgs[0].encrypted_data_size;
+  } else {
+    uint8_t seed[64] = {0};
```

Comment:
Use a macro for `64`

Reply:
we dont have a global macro for seed, should i keep the scope of macro within inheritance?